doc/threat-model.md

```diff
@@ -0,0 +1,9 @@
+Until there is a more concise explanation here about the threat model (what it
+defends against) and assurance case (why it is secure), one should be able to
+get the necessary informaion from this blog post:
+https://cockpit-project.org/blog/is-cockpit-secure.html
+
+The post is not exactly clear on this, but there could be a security boundary
+between cockpit-ws running on one host and another host where cockpit-bridge is
+spawned via ssh, where if the second host was infected with malware it should
+be prevented from harming the first.
```

Comment:
I'm not satisfied with this last sentence, because it doesn't give a decision. But I'm not in a situation to suggest one way or another. Nor have I reviewed the security.

Any suggestion if this should say it currently is or is not treated as a security boundary?

 @mmartinv I think we talked about this?

Reply:
With directly logging into a remote host from the login page, there is a strong security boundary, exactly the same as with `ssh remotehost`. I.e. a malicious remotehost can't influence the source host.

However, this completely breaks down when using the "Add new host" functionality. This is somewhere between "extremely hard" and "impossible" to fix, so we are now considering to entirely remove that functionality. See https://issues.redhat.com/browse/COCKPIT-870 for details. Until we figure out how to do that without breaking existing customers, the text here should point out this distinction.